src/components/EnvelopeList.vue

```diff
@@ -1,25 +1,58 @@
 <template>
 	<div>
+		<transition-group name="multiselect-header">
+			<div v-if="selectMode" key="multiselect-header" class="multiselect-header">
+				<div>
+					<b>{{ selection.length + ' selected' }}</b>
+				</div>
+				<Actions class="app-content-list-item-menu" menu-align="right">
+					<ActionLink href="#" icon="icon-close" @click.prevent="unselectAll">
+						{{ t('mail', 'Unselect') }}
+					</ActionLink>
+				</Actions>
+				<Actions class="app-content-list-item-menu" menu-align="right">
+					<ActionButton icon="icon-mail" @click.prevent="markSelectedSeenOrUnseen(false)">
+						{{ t('mail', 'Mark selected read') }}
+					</ActionButton>
+					<ActionButton icon="icon-mail" @click.prevent="markSelectedSeenOrUnseen(true)">
+						{{ t('mail', 'Mark selected unread') }}
+					</ActionButton>
+					<ActionButton icon="icon-delete" @click.prevent="deleteAllSelected">
+						{{ t('mail', 'Delete selected') }}
+					</ActionButton>
+				</Actions>
+			</div>
+		</transition-group>
 		<transition-group name="list">
 			<div id="list-refreshing" key="loading" class="icon-loading-small" :class="{refreshing: refreshing}" />
 			<Envelope
 				v-for="env in envelopes"
 				:key="env.uid"
 				:data="env"
 				:folder="folder"
+				:selected="env.flags.selected"
```

Comment:
as noted before this should be local state. like an array or object in this component with all selected items. and here you pass that info to the message.

what you're currently do is
1) use global state
2) mutate Vuex state outside a mutation. Vue should already warn you about that

Reply:
ok resolved